react/utils/normalize.ts

```diff
@@ -67,14 +77,149 @@ export function changeImageUrlSize(
+  }
+
+  return itemPrice
+}
+
+/**
+ * @description
+ * Retrieves the SKU with the best price considering a provided condition
+ * @param items an array of SKUs
+ * @param condition the condition [highest|lowest]
+ * */
+function findSKUByPrice(
+  items: ProductTypes.Item[],
+  condition: PriceConditionRule
+) {
+  // If none or only 1 sku is available, avoid reducing
+  const availableItems = items.filter(findAnyAvailable)
+
+  if (availableItems.length === 0) return items[0]
+  if (availableItems.length === 1) return availableItems[0]
+
+  const bestPrices = availableItems.map((item) => getBestPrice(item, condition))
+  let itemToReturn
+
+  if (condition === 'highest') {
+    itemToReturn = bestPrices.reduce(
+      (maxIndex, currentPrice, currentIndex) =>
+        currentPrice > bestPrices[maxIndex] ? currentIndex : maxIndex,
+      0
+    )
+  } else {
+    itemToReturn = bestPrices.reduce(
+      (minIndex, currentPrice, currentIndex) =>
+        currentPrice < bestPrices[minIndex] ? currentIndex : minIndex,
+      0
+    )
+  }
+
+  return availableItems[itemToReturn]
+}
+
+/**
+ * @summary
+ * Used to return the first available SKU
+ * */
+function getAvailableProduct({ sellers }: ProductTypes.Item) {
+  return sellers.find(isAvailable)
+}
+
+/**
+ * @summary
+ * Used to probe if at least 1 seller has stock
+ * */
+function findAnyAvailable({ sellers }: ProductTypes.Item) {
+  return sellers.some(isAvailable)
+}
+
+/**
+ * @description
+ * Attempts to retrieve the specific SKU using a custom Product (field)
+ * in the Catalog. The specification has the value of the SKU ID
+ * that will be looking for. If not found by any reason, we default
+ * to finding the preferred SKU
+ * @param items an array of SKUs
+ * @param defaultSKUspec the ID used to find the SKU
+ * @param preferenceFallback the logic for finding an SKU if the ID fails
+ * @see {@link https://vtex.io/docs/recipes/all} recipe on how to implement this
+ * */
+function findDefaultSKU(
+  items: ProductTypes.Item[],
+  defaultSKUspec: string[],
+  preferenceFallback: PreferenceType
+) {
+  const specificSKU = items.filter(
+    ({ itemId }) => String(itemId) === String(defaultSKUspec)
   )
+
+  if (specificSKU.length) {
+    return specificSKU[0]
+  }
+
+  return findPreferredSKU(items, preferenceFallback)
+}
+
+/**
+ * @description
+ * If there's more than one `Item`, and if there's no defaultSKU, this will
+ * retrieve a suitable one depending on the logic provided
+ * @param items an array of SKUs
+ * @param preferredSKU the logic for finding an SKU
+ * */
+function findPreferredSKU(
+  items: ProductTypes.Item[],
+  preferredSKU: PreferenceType
+) {
+  switch (preferredSKU) {
+    default:
+    case 'FIRST_AVAILABLE':
+      return items.find(getAvailableProduct) ?? items[0]
+
+    case 'LAST_AVAILABLE':
+      return (
+        [...items].reverse().find(getAvailableProduct) ??
+        [...items].reverse()[0]
```

Comment:
What do you think about these suggestions?
```suggestion
        items[items.length - 1]
```
or 
```ts
case 'LAST_AVAILABLE':
    const reservedItems = [...items].reverse()
    return (reversedItems.find(getAvailableProduct) ?? reversedItems[0])
```

Reply:
Added